codegen/core/src/main/java/software/amazon/smithy/python/codegen/PythonFormatter.java

```diff
@@ -72,6 +72,7 @@ private void format(FileManifest fileManifest) {
             return;
         }
         LOGGER.info("Running code formatter on generated code");
+        CodegenUtils.runCommand("python3 -m ruff check --fix", fileManifest.getBaseDir());
```

Comment:
Hmmm... Instead of having ruff fix it, can we instead fix the codegen itself? In other words, have ruff fail the build if, say, unused imports.

Reply:
The problem is that there's not a good way to have section-conditional imports. That would be a big change, and it'd have to go in smithy itself